src/shared/sql.ts

```diff
@@ -36,10 +36,20 @@ export async function getPluginConfigRows(server: PluginsServer): Promise<Plugin
 export async function setError(
     server: PluginsServer,
     pluginError: PluginError | null,
-    pluginConfig: PluginConfig | PluginConfigId
+    pluginConfig: PluginConfig
 ): Promise<void> {
     await server.db.postgresQuery('UPDATE posthog_pluginconfig SET error = $1 WHERE id = $2', [
         pluginError,
         typeof pluginConfig === 'object' ? pluginConfig?.id : pluginConfig,
     ])
+    if (pluginError && server.ENABLE_PERSISTENT_CONSOLE) {
+        await server.db.createPluginLogEntry(
+            pluginConfig,
+            PluginLogEntryType.Error,
+            true,
```

Comment:
Hmm.. Maybe I'm wrong, but I think most of the time this function is called with not "is_system" errors, but just uncaught errors from within the plugin. I don't think those should be system logs/errors, but just plugin errors. System logs are indeed the "Plugin unloaded" type things from below...

Reply:
Makes, sense I replaced this with a varchar choice field `source`, which can be "CONSOLE", "SYSTEM", or "PLUGIN".